File: monotonic-array-896/solution.js
/**
 * LeetCode problem # 896, 'Monotonic Array'
 * https://leetcode.com/problems/monotonic-array/
 */

/**
 * @param {number[]} nums
 * @return {boolean}
 */
function isMonotonic(nums) {
  let isIncreasing = true;
  let isDecreasing = true;

  const n = nums.length;
  for (let i = 1; i < nums.length; ++i) {
    if (nums[i - 1] > nums[i]) {
      isIncreasing = false;
      break;
    }
  }
  for (let i = 1; i < nums.length; ++i) {
    if (nums[i - 1] < nums[i]) {
      isDecreasing = false;
      break;
    }
  }
  return isDecreasing || isIncreasing;
}

File: monotonic-array-896/Solution.java
/**
 * LeetCode problem # 896, 'Monotonic Array'
 * https://leetcode.com/problems/monotonic-array/
 */

class Solution {
    public boolean isMonotonic(int[] nums) {
        return isIncreasing(nums) || isDecreasing(nums);
    }

    private boolean isIncreasing(int[] nums) {
        boolean res = true;
        for (int i = 1; i < nums.length; ++i) {
            if (nums[i - 1] > nums[i]) {
                return false;
            }
        }
        return res;
    }

    private boolean isDecreasing(int[] nums) {
        boolean res = true;
        for (int i = 1; i < nums.length; ++i) {
            if (nums[i - 1] < nums[i]) {
                return false;
            }
        }
        return res;
    }
}